pymeasure/instruments/agilent/agilentB2980.py

```diff
@@ -0,0 +1,132 @@
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from pymeasure.instruments import SCPIMixin, Instrument
+from pymeasure.instruments.validators import truncated_range, strict_discrete_set
+
+class AgilentB298xAmmeter(SCPIMixin, Instrument):
+    """
+    Agilent/Keysight B298xA/B series, Femto/Picoammeter functions.
+    """
+    input_enabled = Instrument.control(
+        ":INP?", ":INP %d",
+        """Control the instrument input (boolean).""",
+        validator=strict_discrete_set,
+        map_values=True,
+        values={True: 1, False: 0},        
+        )
+
+    zero_correction =  Instrument.control(
+        ":INP:ZCOR?", ":INP:ZCOR %d",
+        """
+        Control the zero correct function for current or charge measurement (boolean).
+        
+        B2981/B2983 supports current measurement only.
+        """,
+        validator=strict_discrete_set,
+        map_values=True,
+        values={True: 1, False: 0},        
+        )
+
+class AgilentB298xElectrometer(Instrument):
+    """
+    Agilent/Keysight B298xA/B series, Electrometer/High Resistance Meter functions.
+    """
+    output_enabled = Instrument.control(
+        ":OUTP?", ":OUTP %d",
+        """Control the instrument source output (boolean).""",
+        validator=strict_discrete_set,
+        map_values=True,
+        values={True: 1, False: 0},        
+        )
+
+class AgilentB298xBattery(Instrument):
+    """
+    Battery functions of the B2983/7 models.
+    """
+    battery_level = Instrument.measurement(
+        ":SYST:BATT?",
+        """Get the percentage of the remaining battery capacity.""",
+    )
+    
+    battery_cycles = Instrument.measurement(
+        ":SYST:BATT:CYCL?",
+        """Get the battery cycle count.""",
+    )
+    
+    battery_selftest =  Instrument.measurement(
+        ":SYST:BATT:TEST?",
+        """Self-test off the battery.
+        
+            0: passed
+            1: failed
+        """,
+    )
+
+##########################
+# Instrument definitions #
+##########################
+class AgilentB2981(AgilentB298xAmmeter):
+    """
+    Agilent/Keysight B2981A/B series, Femto/Picoammeter.
+    """
+    def __init__(self, adapter, name="Agilent/Keysight B2981A/B Femto/Picoammeter", **kwargs):
```

Comment:
I'm not against removing these init methods (you still need a `pass` statement).
That way, they all have the same name, but that's not that important in an application.